### ISSUE TITLE
[bug] Incorrect AWS URL formatting in `BuildAwsUrl`

### ISSUE THREAD

**Issue by simon-fauconnier** [minio/minio-cpp#171]
**Description**:

Hello,

I am encountering an issue with AWS URL formatting when using the `minio-cpp` client. Following the `BucketExists` example provided in the documentation, an error is raised regarding host resolution.

### Code to reproduce the issue

```cpp
#include <miniocpp/client.h>

int main() {
  // Create S3 base URL.
  minio::s3::BaseUrl base_url("s3.eu-central-1.amazonaws.com");

  // Create credential provider.
  minio::creds::StaticProvider provider(
      "****", "****");

  // Create S3 client.
  minio::s3::Client client(base_url, &provider);

  // Create bucket exists arguments.
  minio::s3::BucketExistsArgs args;
  args.bucket = "test";

  // Call bucket exists.
  minio::s3::BucketExistsResponse resp = client.BucketExists(args);

  // Handle response.
  if (resp) {
    if (resp.exist) {
      std::cout << "my-bucket exists" << std::endl;
    } else {
      std::cout << "my-bucket does not exist" << std::endl;
    }
  } else {
    std::cout << "unable to do bucket existence check; "
              << resp.Error().String() << std::endl;
  }

  return 0;
}
```

### Debugging Output

When `client.Debug(true);` is enabled, the following output is observed:

```
* Could not resolve host: sa-soundnodes.eu-central-1.amazonaws.comeu-central-1.amazonaws.com
* Closing connection 0
unable to do bucket existence check; server failed with HTTP status code 0
```

It seems that the AWS URL is not formatted correctly, leading to a concatenated host name that cannot be resolved (`sa-soundnodes.eu-central-1.amazonaws.comeu-central-1.amazonaws.com`).

### Expected Behavior

The client should properly format the AWS URL so that the host resolves correctly.

### Comparison with Python Client

I do not have this issue when using the Python client for MinIO. The following Python code works correctly without any host resolution problems:

```python
from minio import Minio

client = Minio("s3.eu-central-1.amazonaws.com",
    access_key="****",
    secret_key="****",
)

bucket_name = "test"

found = client.bucket_exists(bucket_name)
if not found:
    print("Bucket not exists", bucket_name)
else:
    print("Bucket", bucket_name, "already exists")
```

The Python client successfully checks the existence of the bucket without any issues.

### Environment

- `minio-cpp` version: 0.3.0
- Operating system: Ubuntu 22.04.3 LTS
- Compiler: gcc version 11.4.0

**Comment by balamurugana** [comment]
@simon-fauconnier Please try the fix https://github.com/minio/minio-cpp/pull/172

**Comment by simon-fauconnier** [comment]
The fix #172 solves the issue

Thanks @balamurugana !
